include/zephyr/drivers/rtc/rtc_rv3028.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2024 ANITRA system s.r.o.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_RTC_RTC_RV3028_H_
+#define ZEPHYR_INCLUDE_DRIVERS_RTC_RTC_RV3028_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+
+void rv3028_lock_sem(const struct device *dev);
```

Comment:
Really not a fan of having these as global functions, should be used with driver function pointers of the parent device in my mind. Thoughts @henrikbrixandersen ?

Reply:
Yeah, my thoughts exactly. Forgot to ask about this.

Reply:
I agree.

Reply:
@henrikbrixandersen any pointers how to implement it? The parent driver interface will still need to be global, right?